### PR TITLE
server: Correct status codes in several responses.

### LINKED ISSUE #195: Should 'Too many recipients' return 452 4.5.3 instead of 552 5.5.3?
I've been looking into this response a bit for something I'm working on. The current response when we have a recipient limit is

```
552 5.5.3 Maximum limit of %v recipients reached
```

From my reading though, it seems the following is the more 'standard' response for this case

```
452 4.5.3 Too many recipients
```

with some providers including the limit there. Specifically I'm concerned with how the big players say to handle this status code to make sure that they send correctly with fewer recipients. I'm about to do testing with the current code and will report my findings, but I think this should be changed.

**Gmail**: [https://support.google.com/a/answer/3726730](https://support.google.com/a/answer/3726730)
> 452, "4.5.3", Your message has too many recipients. For more information regarding Google's sending limits, visit [Limits for sending & getting mail.](https://support.google.com/mail/answer/22839)

**Microsoft**: [https://support.google.com/a/answer/3726730](https://support.google.com/a/answer/3726730)
> For inbound email, the Receive connector MaxRecipientsPerMessage is verified first. However, if the number of recipients exceeds the limit, the message is not rejected; the connection receives the error, 452 4.5.3 Too many recipients. Most mail servers understand this error and they will continue to resend the message in another connection until the message is delivered to all recipients.

Granted they are talking about their own servers and how they expect clients to handle it, so I would hope they respond the same way. The rfc's may suggest that 5xx is more correct (I don't think they specifically say what to do for too many recipients), but if gmail and ms agree on something else, I would consider it the de-facto standard.

### PR DESCRIPTION
**Pull request by ikedas**
- The excess number of recipients should be responded with `4XX`.
- Disconnections due to network errors or idle timeouts should be `4XX`.
- `3XX` codes do not have enhanced codes.
- Other typos.
---
This PR may fix #195 .

### CODE REVIEW

**Comment by ikedas** [comment]
This PR may fix #195

File: conn.go [hunk 1]
```diff
@@ -467,7 +467,7 @@ func (c *Conn) handleRcpt(arg string) {
 	recipient := strings.Trim(arg[3:], "<> ")
 
 	if c.server.MaxRecipients > 0 && len(c.recipients) >= c.server.MaxRecipients {
-		c.writeResponse(552, EnhancedCode{5, 5, 3}, fmt.Sprintf("Maximum limit of %v recipients reached", c.server.MaxRecipients))
+		c.writeResponse(452, EnhancedCode{4, 5, 3}, fmt.Sprintf("Maximum limit of %v recipients reached", c.server.MaxRecipients))
```
Comment (emersion) [inline conn.go:470]: It seems like the only valid short code for x.5.3 is 451, according to https://www.iana.org/assignments/smtp-enhanced-status-codes/smtp-enhanced-status-codes.xhtml

**Review by emersion — CHANGES_REQUESTED** [review]
LGTM apart from this

(Would've been nice to split the greeting message changes into its own commit.)

File: conn.go (hunk 1 above)
Comment (ikedas) [reply conn.go:470]: I found that [RCF 5321](https://www.rfc-editor.org/rfc/rfc5321#section-4.5.3.1.9) states that basic status code for "Too many recipients" is `452`.  The "Associated Basic Status Code" field in IANA registry does not cover all possible values,

**Comment by ikedas** [comment]
> (Would've been nice to split the greeting message changes into its own commit.)

Makes sense.  I've done.

File: conn.go (hunk 1 above)
Comment (emersion) [reply conn.go:470]: Right!

**Review by emersion — APPROVED** [review]
Thanks!

**Pull request merged**